### DISCUSSION TITLE
Getting more debug information from between pytest call and start of tests?

### DISCUSSION THREAD

**Discussion by behackl** [pytest-dev/pytest#12957] [Q&A] [answered]
Hey there, [over at Manim](https://github.com/ManimCommunity/manim) we are having a hard time getting our test suit to run properly on a GitHub Actions runner with the `windows-latest` tag and Python 3.13 installed. This OS/Python combination is the only one where pytest just exits with code 1 before starting the tests or printing the report header; see https://github.com/ManimCommunity/manim/actions/runs/11722294226/job/32651526951 for the logs from one of these runs.

Unfortunately, I can't really reproduce this on a local device, so I am more or less stuck with debugging using the runner. I've tried [quite a bunch of things](https://github.com/ManimCommunity/manim/pull/3967) that I could think of to get more detailed information -- for example, here is a run of `python -v -m pytest`: https://github.com/ManimCommunity/manim/actions/runs/11770345647/job/32782653900 -- but nothing really helped so far.

In my latest attempt I tried to trigger consistent pytest errors (during testing) by removing one of the fixtures we use, see https://github.com/ManimCommunity/manim/actions/runs/11788149689/job/32834760283 -- all OS / python version combinations behave as expected, except for our pathological case where the runners appear to experience hard crashes. Something about this combination seems quite cursed.

I've not found anything explicit in pytest's documentation (but I might have overlooked it, of course): is there any way to get more debug information for the case where pytest exits prematurely? Or is this perhaps a known issue?

I'd be very happy and grateful for any sort of general pointers; thank you very much for your efforts!

**Comment by RonnyPfannschmidt** [comment, marked as answer]
The `PYTEST_DEBUG` environment var triggers excessive debug output 

The flag combination `-sv` may suffice to pin it down 

Also, take a look at the built-in faulthandler plugin

**Reply by behackl** [reply]
Thanks! While I still don't know the exact reason for the problem, `PYTEST_DEBUG` has helped to at least confirm that one of our modules seems to be responsible; it seems to happen in the step leading to `early skip of rewriting module: manim.constants [assertion]`. This gives us something to work with.
